pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -118,16 +118,32 @@ func GetVerifiedPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekt
 			source = s.URI
 		}
 		logger := logging.FromContext(ctx)
-		if config.CheckEnforceResourceVerificationMode(ctx) || config.CheckWarnResourceVerificationMode(ctx) {
-			if err := trustedresources.VerifyPipeline(ctx, p, k8s, source, verificationpolicies); err != nil {
-				if config.CheckEnforceResourceVerificationMode(ctx) {
-					logger.Errorf("GetVerifiedPipelineFunc failed: %v", err)
-					return nil, nil, fmt.Errorf("GetVerifiedPipelineFunc failed: %w: %v", trustedresources.ErrResourceVerificationFailed, err)
-				}
+		// skip the verification if no policies are found and trusted-resources-verification-no-match-policy is set to ignore
+		if len(verificationpolicies) == 0 {
+			if config.CheckWarnVerificationNoMatchPolicy(ctx) {
+				// log warning and don't return error when trusted-resources-verification-no-match-policy is set to warn
 				logger.Warnf("GetVerifiedPipelineFunc failed: %v", err)
 				return p, s, nil
 			}
+			if config.CheckAllowVerificationNoMatchPolicy(ctx) {
+				return p, s, nil
+			}
+		}
+
+		if err := trustedresources.VerifyPipeline(ctx, p, k8s, source, verificationpolicies); err != nil {
+			if errors.Is(err, trustedresources.ErrNoMatchedPolicies) && !config.CheckDenyVerificationNoMatchPolicy(ctx) {
+				if config.CheckWarnVerificationNoMatchPolicy(ctx) {
+					// log warning and don't return error when trusted-resources-verification-no-match-policy is set to warn
+					logger.Warnf("GetVerifiedPipelineFunc failed: %v", err)
+				}
+				// don't return error if no matching policies are found and trusted-resources-verification-no-match-policy is set to allow
+				return p, s, nil
+			}
```

Comment:
What happens if the config is forced into a configuration where it's not one of the 3 expected values?

This might be better if this was structured like:

```go
if errors.Is(err, trustedresources.ErrNoMatchedPolicies) {
 switch config.GetVerificationNoMatchPolicy(ctx) {
    case Deny:
      ...
    case Warn:
      ...
    case Allow:
      ...
    default:
      ...
  }
}

Reply:
ah yes, I think this is better! Thanks!